### PR TITLE
Fixed ConsolidateBlocks over multiple registers

### LINKED ISSUE #2806: ConsolidateBlocks raises for CX between two registers
From https://travis-ci.com/Qiskit/qiskit-terra/jobs/216588160#L6863:

```
>>> qr1 = qk.QuantumRegister(1)
>>> qr2 = qk.QuantumRegister(2)
>>> qc = qk.QuantumCircuit(qr2, qr1, cr)
>>> qc.cx(qr1[0], qr2[1])
>>> qc.measure(qr1[0], cr[0])
>>> qk.transpile(qc, optimization_level=3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kevin.krsulichibm.com/q/qiskit-terra/qiskit/compiler/transpile.py", line 187, in transpile
    circuits = parallel_map(_transpile_circuit, list(zip(circuits, transpile_configs)))
  File "/Users/kevin.krsulichibm.com/q/qiskit-terra/qiskit/tools/parallel.py", line 100, in parallel_map
    return [task(values[0], *task_args, **task_kwargs)]
  File "/Users/kevin.krsulichibm.com/q/qiskit-terra/qiskit/compiler/transpile.py", line 208, in _transpile_circuit
    return transpile_circuit(circuit, transpile_config)
  File "/Users/kevin.krsulichibm.com/q/qiskit-terra/qiskit/transpiler/transpile_circuit.py", line 65, in transpile_circuit
    return pass_manager.run(circuit)
  File "/Users/kevin.krsulichibm.com/q/qiskit-terra/qiskit/transpiler/passmanager.py", line 171, in run
    dag = self._do_pass(pass_, dag, passset.options)
  File "/Users/kevin.krsulichibm.com/q/qiskit-terra/qiskit/transpiler/passmanager.py", line 202, in _do_pass
    dag = self._run_this_pass(pass_, dag)
  File "/Users/kevin.krsulichibm.com/q/qiskit-terra/qiskit/transpiler/passmanager.py", line 215, in _run_this_pass
    new_dag = pass_.run(dag)
  File "/Users/kevin.krsulichibm.com/q/qiskit-terra/qiskit/transpiler/passes/consolidate_blocks.py", line 93, in run
    subcirc.append(nd.op, [q[block_index_map[i]] for i in nd.qargs])
  File "/Users/kevin.krsulichibm.com/q/qiskit-terra/qiskit/circuit/quantumcircuit.py", line 359, in append
    instructions.add(self._append(instruction, qarg, carg), qarg, carg)
  File "/Users/kevin.krsulichibm.com/q/qiskit-terra/qiskit/circuit/quantumcircuit.py", line 382, in _append
    self._check_dups(qargs)
  File "/Users/kevin.krsulichibm.com/q/qiskit-terra/qiskit/circuit/quantumcircuit.py", line 438, in _check_dups
    raise QiskitError("duplicate qubit arguments")
qiskit.exceptions.QiskitError: 'duplicate qubit arguments'
```

### PR DESCRIPTION
**Pull request by maddy-tod**
### Summary
Fixes #2806 

The issue was the way registers were being indexed in the pass. Previously, the overall index of a qubit was `overall_index_of_reg + index_of_wire_in_reg`. This became an issue when the registers had a size larger than 1.

For example,
```
 qr0 = QuantumRegister(2, "qr0")
 qr1 = QuantumRegister(1, "qr1") 
```
would lead to a global list of registers `[qr0, qr1]`. This would then index the qubits using the above formula as
```
qr0[0] = 0 + 0 = 0
qr0[1] = 0 + 1 = 1
qr1[0] = 1 + 0 = 1
```
As you can see this means there are 2 qubits with the overall index of `1` and so when a `CX` gate was applied between these 2 it appeared to be applied to the same qubit twice, the qubit at index `1`

### Details and comments
I fixed this by updating the way the index was calculated. 
It is now `sum_of_all_prev_reg_sizes + index_of_wire_in_reg`

### CODE REVIEW

File: qiskit/transpiler/passes/consolidate_blocks.py [hunk 1]
```diff
@@ -62,7 +62,16 @@ def run(self, dag):
...
+
+            reg_index = 0
+            for g_qreg in global_qregs:
+                # if this is the register we want, stop
+                if g_qreg == wire.register:
+                    break
+                # index needs to increase by the size of the register
+                reg_index += g_qreg.size
+
+            global_index_map[wire] = reg_index + wire.index
```
Comment (kdk) [inline qiskit/transpiler/passes/consolidate_blocks.py]: To make sure I understand, it looks like `global_index_map` should be a mapping from every `Qubit` wire in `dag.wires` to an integer, its index in the ordered list of (register, wire) pairs, set by the register ordering in `dag.qregs.values()` and the index ordering of the wire within its register. Is that right?

If so, could this be simplified by laying out all the `Qubit` wires in the desired order first and building the mapping from the indices of that list? Maybe something like `global_index_map = { wire, idx for idx, wire in enumerate(dag.qubits()) }` would be equivalent?

Comment (maddy-tod) [reply qiskit/transpiler/passes/consolidate_blocks.py]: Yes, you are correct. I see that `dag.qubits()` uses the `qregs.values()` ordering so this should be equivalent.

Comment (maddy-tod) [reply qiskit/transpiler/passes/consolidate_blocks.py]: Updated in cbfd528

**Comment by kdk** [comment]
Closing and re-opening to trigger azure builds.

**Comment by mtreinish** [comment]
Trying that one more time

**Comment by mtreinish** [comment]
Azure is still acting up, but it did manage to run all jobs successfully, it just did not report back as such. You can see the results for `Lint_and_Tests Linux_Tests Python35` which is still listed as pending here: https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=1714&view=logs&jobId=b59abcfc-1a00-5817-c1e4-d93e425db181&taskId=5e8156b8-6f6a-5ddc-c9a9-7a100c55c902&lineStart=2109&lineEnd=2110&colStart=1&colEnd=1 so going ahead with merging this.

**Pull request merged**